src/index.ts

```diff
@@ -1,4 +1,150 @@
-export default () => {
-  // eslint-disable-next-line no-console
-  console.log('https://www.youtube.com/watch?v=dQw4w9WgXcQ')
+import ResponseError from './ResponseError'
+
+export interface ApiClientConfiguration {
+  baseUrl: string
+  headers: Record<string, string>
 }
+
+export interface RequestOptions {
+  baseUrl?: string
+  headers?: Record<string, string>
+  params?: Record<string, string | number>
+  accessToken?: string
+}
+
+class ApiClient {
+  // eslint-disable-next-line no-use-before-define
+  private static instance: ApiClient
+
+  private configuration: ApiClientConfiguration
```

Comment:
Can we define the value of `configuration` directly instead of `constructor`. It should perform the same behavior

```ts
private configuration: ApiClientConfiguration = {
    baseUrl: '',
    headers: { 'Content-Type': 'application/json' },
}
```